deepspeed/module_inject/replace_module.py

```diff
@@ -373,6 +373,25 @@ def _replace(child, name, conv_linear_layer):
                 return
             mp_replace = ReplaceWithTensorSlicing(mp_group=mp_group)
             weight_shape = child.weight.shape
+            if name == 'attn.Wqkv' and module._get_name() == 'MPTBlock':
+                # MPT block qkv weight's allocation is different from other models, it's [3,num_head,head_dim,head_dim]
```

Comment:
just curious do you mean [3, num_head, dim_per_head, hidden_size(=num_head*dim_per_head)] ? thanks

Reply:
yes. I will fix the comment, thanks

Reply:
hi, @sywangyi  I have sent a PR for fix different types of fusedqkv weight last week. Including this type. 
https://github.com/microsoft/DeepSpeed/pull/3844/files

Reply:
yes, I could leverage some change in https://github.com/microsoft/DeepSpeed/pull/3844/files, will update PR once 3844 is merged.